site/blog/2022/deploy-static-website.mdx

```diff
@@ -0,0 +1,193 @@
+---
+slug: deploy-static-website
+title: Deploy a Static Website Using SQL
+date: 2022-12-21
+unlisted: true
+authors: [mtp1376]
+---
+Did you know you can deploy a static website without leaving your SQL REPL? In this post we'll show you how to leverage IaSQL's capabilities to simply deploy a static website from your GitHub repository to AWS using only SQL commands.
+
+## Overview
+First, we will create a S3 bucket which will hold our static files. Then we'll enable S3 static website hosting to be able to access our website using a S3 endpoint. We'll also need to enable public access to our S3 files. After that we will create a new CloudFront distribution because S3 static website hosting doesn't support HTTPS, and we're done. Let's dive in to see how IaSQL can deploy a static website using simple SQL commands.
+
+## Create a S3 Bucket
+To be able to work with S3, we should first install corresponding IaSQL module.
+```sql TheButton[Install S3 Module]="Install S3 Module"
+select iasql_install('aws_s3');
+```
+
+Installing a module gives us ability to use tables and RPCs provided by it. `aws_s3` module gives us ability to manage a S3 bucket, S3 static website hosting and other related stuff. So let's create a S3 bucket first.
+```sql TheButton[Create a S3 bucket]="Create a S3 Bucket"
+SELECT iasql_begin();
+INSERT INTO bucket (name, policy_document, region) VALUES ('<bucket-name>', '{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "PublicReadGetObject",
+      "Effect": "Allow",
+      "Principal": "*",
+      "Action": [
+        "s3:GetObject"
+      ],
+      "Resource": [
+        "arn:aws:s3:::<bucket-name>/*"
+      ]
+    }
+  ]
+}', 'us-east-1');
+SELECT iasql_commit();
+```
+The above command will direct IaSQL to create a new bucket in the `us-east-1` region with the name `<bucket-name>` and with the given policy. The `iasql_begin` and `iasql_commit` functions are RPCs that will start and finish an IaSQL transaction. For more info on this topic you can read [this doc](https://iasql.com/docs/next/transaction/).
```

Comment:
is there any restriction in the policy that needs to be used?

Reply:
It makes the `s3:GetObject` available to all the objects in the bucket.